tools/pipelines-tasks/common/nuget_helper.ts

```diff
@@ -0,0 +1,83 @@
+import * as path from "path";
+import * as tl from "azure-pipelines-task-lib/task";
+import * as fs from 'fs';
+import fetch from 'node-fetch';
+
+/**
+ * downloads nuget tool(nuget.exe).
+ * @param url weblink to download nuget tool
+ * @param downloadLoc location to download the nuget too into
+ */
+export async function downloadNugetTool(url: string, downloadLoc: string): Promise<void> {
+    try {
+        const response = await fetch(url);
+
+        if (!response.ok) {
+            throw new Error(`Failed to download file. Status: ${response.status} - ${response.statusText}`);
+        }
+
+        const fileStream = fs.createWriteStream(path.join(downloadLoc, 'nuget.exe'));
+        await new Promise<void>((resolve, reject) => {
+            response.body.pipe(fileStream);
+            response.body.on('error', (err: any) => {
+                reject(err);
+            });
+            fileStream.on('finish', function () {
+                resolve();
+            });
+        });
+    } catch (error) {
+        console.error("Error downloading file: " + error);
+    }
+}
+
+/**
+ * picks installed nuget and copies target DLLs into output folder
+ * @param folderPath the path to the package(.nupkg or packages.config) file to be installed
+ * @param targetNetFramework target .Net framework to pick target and dependency DLLs complying with the package to be installed
```

Comment:
it is helpful. Say, we update target framework for app attach framework in future, it would help.

Reply:
I had understood that this was copying the DLLs for just the app attach framework. Does it? Or is it copying DLLs for all the dependencies, too? If that's the case, I can understand why keep it. But if it is only for app attach framework, I think it would be easier to just update the folder path whenever we update target framework, rather than maintaining this function.

Reply:
This is for dependent DLLs along with appatttachframework DLL